modules/core/arrow-test/src/main/kotlin/arrow/test/laws/AsyncLaws.kt

```diff
@@ -163,6 +170,31 @@ object AsyncLaws {
         p.equalUnderTheLaw(just(true), EQK)
     }
 
+  fun <F> Async<F>.derivedDefer(GK: GenK<F>, EQK: Eq<Kind<F, Int>>): Unit =
+    forAll(Gen.functionToA(GK.genK(Gen.int()))) { f: () -> Kind<F, Int> ->
+      defer(one, f).equalUnderTheLaw(just(Unit).continueOn(one).flatMap { defer(f) }, EQK)
+    }
+
+  fun <F> Async<F>.derivedLaterOrRaise(EQK: Eq<Kind<F, Int>>): Unit =
+    forAll(Gen.functionToA(Gen.either(Gen.throwable(), Gen.int()))) { f: () -> Either<Throwable, Int> ->
+      laterOrRaise(one, f).equalUnderTheLaw(defer(one) { f().fold({ raiseError<Int>(it) }, { just(it) }) }, EQK)
+    }
+
+  fun <F> Async<F>.derivedContinueOn(EQK: Eq<Unit>) {
+    fx.async {
+      continueOn(one).equalUnderTheLaw(one.shift().bind(), EQK)
+    }
```

Comment:
here nothing is ever executed, you need to make sure you either assert something or call equalUnderTheLaw as the last statement:

```
fun <F> Async<F>.derivedContinueOn(EQK: Eq<Kind<F, Unit>>) {
  fx.async { continueOn(one) }.equalUnderTheLaw(fx.async { one.shift().bind() }, EQK)
}
```